alphagov/paas-cf
# [#154524416] Rename paas-usage-events-collector to paas-billing

## What

Rename paas-usage-events-collector to paas-billing

The old name was difficult to remember and didn't reflect the fact that
this app now has an API to reporting billing.

Some FIXMEs are included for tidying up.

I haven't attempted to redirect the old hostname because we are the only
users and we can communicate the change on our team mailing list.

## How to review

1. Deploy from `master` with:

        SELF_UPDATE_PIPELINE=false ENABLE_USAGE_EVENTS_COLLECTION=true make dev pipelines
1. Run the `create-cloudfoundry` pipeline.
1. Confirm that you can access and login to:

    - https://paas-usage-events-collector.${DEPLOY_ENV}.dev.cloudpipelineapps.digital
1. Deploy from this branch with:

        git fetch && git checkout bugfix/154524416-rename_to_billing
        BRANCH=$(git rev-parse --abbrev-ref HEAD) SELF_UPDATE_PIPELINE=false ENABLE_BILLING_APP=true make dev pipelines
1. Run the `create-cloudfoundry` pipeline again.
1. Confirm that the old app name no longer exists in the `admin/billing` space.
1. Confirm that the old UAA client has been removed (looking at the build output should be enough).
1. Confirm that you can access and login to:

    - https://paas-billing.${DEPLOY_ENV}.dev.cloudpipelineapps.digital

🐝 🐝 🐝 **fore merging:** 🐝 🐝 🐝 

1. Merge alphagov/paas-release-ci#52
1. Merge alphagov/paas-billing#10
1. Wait for CI to build a new tag.
1. Rebase this branch to remove the `tmp` commit.

## Who can review

Anyone with access to Build Concourse in CI.
